packages/web-components/fast-foundation/src/data-grid/data-grid.ts

```diff
@@ -0,0 +1,608 @@
+
+    /**
+     * The template used to render header cells in generated rows.
+     *
+     * @public
+     */
+    @observable
+    public headerCellItemTemplate?: ViewTemplate;
+    private headerCellItemTemplateChanged(): void {
+        if (this.$fastController.isConnected) {
+            if (this.generatedHeader !== null) {
+                this.generatedHeader.headerCellItemTemplate = this.headerCellItemTemplate;
+            }
+        }
+    }
+
+    /**
+     * The index of the row that will receive focus the next time the
+     * grid is focused. This value changes as focus moves to different
+     * rows within the grid.  Changing this value when focus is already
+     * within the grid moves focus to the specified row.
+     *
+     * @public
+     */
+    @observable
+    public focusRowIndex: number = 0;
+    private focusRowIndexChanged(): void {
+        if (this.$fastController.isConnected) {
+            this.queueFocusUpdate();
+        }
+    }
+
+    /**
+     * The index of the column that will receive focus the next time the
+     * grid is focused. This value changes as focus moves to different rows
+     * within the grid.  Changing this value when focus is already within
+     * the grid moves focus to the specified column.
+     *
+     * @public
+     */
+    @observable
+    public focusColumnIndex: number = 0;
+    private focusColumnIndexChanged(): void {
+        if (this.$fastController.isConnected) {
+            this.queueFocusUpdate();
+        }
+    }
+
+    /**
+     * The default cell item template.  Set by the component templates.
+     *
+     * @internal
+     */
+    @observable
+    public defaultRowItemTemplate: ViewTemplate;
+
+    /**
+     * Children that are rows
+     *
+     * @internal
+     */
+    @observable
+    public rowElements: HTMLElement[];
+
+    private rowsRepeatBehavior: RepeatBehavior | null;
+    private rowsPlaceholder: Node | null = null;
+
+    private generatedHeader: DataGridRow | null = null;
+
+    private isUpdatingFocus: boolean = false;
+    private pendingFocusUpdate: boolean = false;
+
+    private observer: MutationObserver;
+
+    private rowindexUpdateQueued: boolean = false;
+    private columnDefinitionsStale: boolean = true;
+
+    private generatedGridTemplateColumns: string = "";
+
+    constructor() {
+        super();
+    }
+
+    /**
+     * @internal
+     */
+    public connectedCallback(): void {
+        super.connectedCallback();
+
+        if (this.rowItemTemplate === undefined) {
+            this.rowItemTemplate = this.defaultRowItemTemplate;
+        }
+
+        this.rowsPlaceholder = document.createComment("");
+        this.appendChild(this.rowsPlaceholder);
+
+        this.toggleGeneratedHeader();
+
+        this.rowsRepeatBehavior = new RepeatDirective(
+            x => x.rowsData,
+            x => x.rowItemTemplate,
+            { positioning: true }
+        ).createBehavior(this.rowsPlaceholder);
```

Comment:
Would this be the same as:

```suggestion
        this.rowsRepeatBehavior = repeat(
            x => x.rowsData,
            x => x.rowItemTemplate,
            { positioning: true }
        ).createBehavior(this.rowsPlaceholder);
```